src/lua/digest.lua

```diff
@@ -267,4 +313,70 @@ m['aes256cbc'] = {
     end
 }
 
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+
+    local function update(self, str)
+        if type(str) ~= 'string' then
+            error("Usage xxhash" .. var .. ":update(string)", 2)
+        end
+        return builtin[update_fn_name](self.value, str, #str) == builtin.XXH_OK
+    end
+
+    local function result(self)
+        return builtin[digest_fn_name](self.value)
+    end
+
+    local function clear(self, seed)
+        if seed == nil then
+            seed = self.default_seed
+        end
+        return builtin[reset_fn_name](self.value, seed) == builtin.XXH_OK
+    end
+
+    local function copy(self)
+        local copy = xxHash.new()
+        builtin[copy_fn_name](copy.value, self.value)
+        return copy
+    end
+
+    local XXH_state_t = ffi.typeof('XXH' .. var .. '_state_t')
+    local default_seed = 0
```

Comment:
But it is not much different from what was before. `default_seed` is an upvalue caught in the `new` closures below. You just need to inline it. Otherwise it will be an upvalue regardless of what you do. Although there is a tiny change the Lua interpreter somehow inlines it if detects it is never changed, however I wouldn't rely on it.

Reply:
Ok, fixed